lstchain/tools/lstchain_create_irf_files.py

```diff
@@ -0,0 +1,399 @@
+        "config": "DataSelection.config",
+        "overwrite": "IRFFITSWriter.overwrite",
+    }
+
+    flags = {
+        "point-like": (
+            {"IRFFITSWriter": {"point_like": True}},
+            "Full Enclosure IRFs will be produced",
+        ),
+        "overwrite": (
+            {"IRFFITSWriter": {"overwrite": True}},
+            "overwrites output file",
+        )
+    }
+
+    def setup(self):
+
+        if self.output_irf_file.absolute().exists():
+            if self.overwrite:
+                self.log.warning(f"Overwriting {self.output_irf_file}")
+                self.output_irf_file.unlink()
+            else:
+                raise ToolConfigurationError(
+                    f"Output file {self.output_irf_file} already exists,"
+                    " use --overwrite to overwrite"
+                )
+
+        filename = self.output_irf_file.name
+        if filename.split(".")[1] != "fits":
+            self.log.warning(
+                f"{filename} is not a correct "
+                "compressed FITS file name. It will be corrected."
+            )
+            filename = filename.split(".")[0] + ".fits.gz"
+            self.output_irf_file = self.output_irf_file.parent / filename
+
+        if self.input_proton_dl2 and self.input_electron_dl2 is not None:
+            self.only_gamma_irf = False
+        else:
+            self.only_gamma_irf = True
+
+        self.data_sel = DataSelection(parent=self)
+        self.data_bin = DataBinning(parent=self)
+
+        self.mc_particle = {
+            "gamma": {
+                "file": str(self.input_gamma_dl2),
+                "target_spectrum": CRAB_HEGRA,
+            },
+        }
+        Provenance().add_input_file(self.input_gamma_dl2)
+
+        # Read and update MC information
+        if not self.only_gamma_irf:
+            self.mc_particle["proton"] = {
+                "file": str(self.input_proton_dl2),
+                "target_spectrum": IRFDOC_PROTON_SPECTRUM,
+            }
+
+            self.mc_particle["electron"] = {
+                "file": str(self.input_electron_dl2),
+                "target_spectrum": IRFDOC_ELECTRON_SPECTRUM,
+            }
+
+            Provenance().add_input_file(self.input_proton_dl2)
+            Provenance().add_input_file(self.input_electron_dl2)
+
+        self.provenance_log = self.output_irf_file.parent / (
+            self.name + ".provenance.log"
+        )
+
+    def start(self):
+
+        for particle_type, p in self.mc_particle.items():
+            self.log.info(f"Simulated {particle_type.title()} Events:")
+            p["events"], p["simulation_info"] = read_mc_dl2_to_QTable(p["file"])
+
+            if p["simulation_info"].viewcone.value == 0.0:
+                p["mc_type"] = "point_like"
+            else:
+                p["mc_type"] = "diffuse"
+                # For diffuse gamma using Proton Spectra for calculating event weights
+                if particle_type == "gamma":
```

Comment:
This doesn't make sense. Why would you use the proton spectrum for diffuse gamma-rays?

Why are you calculating event weights at all here for the gammas?

These are not needed for IRFs, only for sensitivity.

Reply:
True, I do not add any sensitivity HDU here, but, in the future, we may have to, so just kept it here before-hand.
For calculating event weights for diffuse gamma, we do need a diffuse spectra right? So, as a work-around, I used proton spectra here.

Reply:
But you don't need weights for diffuse gammas. And calculating weights with the wrong spectrum is certainly wrong.

Reply:
Ok. I will remove the event weight calculation for gammas.